### PR TITLE
Headers structure changing and getSinglePaymentInformation DeutscheBank implementation

### PR DESCRIPTION
**Pull request by n0rdy**
- implemented ApacheHttpClient for get and put requests;
- introduced initiateSinglePayment() method;
- simplified ResponseHandler of HttpClient;
- extracted headers to the single class;
- introduced response for SinglePaymentInformation;
- changed ErrorResponseException to make it work with no body error responses;
- implemented getSinglePaymentInformation for DeutscheBank;
- changed SinglePaymentInformationMapper to be generated by MapStruct.

### CODE REVIEW

File: xs2a-gateway-rest-server/src/main/java/de/adorsys/xs2a/gateway/mapper/SinglePaymentInformationMapper.java [hunk 1]
```diff
@@ -0,0 +1,10 @@
+package de.adorsys.xs2a.gateway.mapper;
+
+import de.adorsys.xs2a.gateway.model.pis.PaymentInitiationSctWithStatusResponse;
+import de.adorsys.xs2a.gateway.service.SinglePaymentInitiationInformationWithStatusResponse;
+import org.mapstruct.Mapper;
+
+@Mapper(uses = {AddressMapper.class, AmountMapper.class, AccountReferenceMapper.class})
+public interface SinglePaymentInformationMapper {
```
Comment (spetrychenko) [inline xs2a-gateway-rest-server/src/main/java/de/adorsys/xs2a/gateway/mapper/SinglePaymentInformationMapper.java]: test is missing

File: xs2a-gateway-rest-server/src/main/java/de/adorsys/xs2a/gateway/mapper/AmountMapper.java [hunk 2]
```diff
@@ -0,0 +1,21 @@
+package de.adorsys.xs2a.gateway.mapper;
+
+import de.adorsys.xs2a.gateway.model.ais.AmountTO;
+import de.adorsys.xs2a.gateway.service.Amount;
+import org.mapstruct.Mapper;
+
+import java.util.Currency;
+
+@Mapper
+public interface AmountMapper {
```
Comment (spetrychenko) [inline xs2a-gateway-rest-server/src/main/java/de/adorsys/xs2a/gateway/mapper/AmountMapper.java]: test is missing

File: xs2a-gateway-rest-server/src/main/java/de/adorsys/xs2a/gateway/mapper/AddressMapper.java [hunk 3]
```diff
@@ -0,0 +1,11 @@
+package de.adorsys.xs2a.gateway.mapper;
+
+import de.adorsys.xs2a.gateway.model.shared.AddressTO;
+import de.adorsys.xs2a.gateway.service.Address;
+import org.mapstruct.Mapper;
+
+@Mapper
+public interface AddressMapper {
```
Comment (spetrychenko) [inline xs2a-gateway-rest-server/src/main/java/de/adorsys/xs2a/gateway/mapper/AddressMapper.java]: test is missing

**Comment by spetrychenko** [comment]
also please export postman collection if you not done it yet

File: xs2a-gateway-rest-server/src/main/java/de/adorsys/xs2a/gateway/mapper/SinglePaymentInformationMapper.java (hunk 1 above)
Comment (n0rdy) [reply xs2a-gateway-rest-server/src/main/java/de/adorsys/xs2a/gateway/mapper/SinglePaymentInformationMapper.java]: Done, thanks!

File: xs2a-gateway-rest-server/src/main/java/de/adorsys/xs2a/gateway/mapper/AmountMapper.java (hunk 2 above)
Comment (n0rdy) [reply xs2a-gateway-rest-server/src/main/java/de/adorsys/xs2a/gateway/mapper/AmountMapper.java]: Done, thanks!

File: xs2a-gateway-rest-server/src/main/java/de/adorsys/xs2a/gateway/mapper/AddressMapper.java (hunk 3 above)
Comment (n0rdy) [reply xs2a-gateway-rest-server/src/main/java/de/adorsys/xs2a/gateway/mapper/AddressMapper.java]: Done, thanks!

**Comment by n0rdy** [comment]
Postman collection export will be done in a separate PR - name convention has to be discussed.

**Pull request merged**